twine/repository.py

```diff
@@ -67,6 +70,9 @@ def __init__(
         self.session.auth = (
             (username or "", password or "") if username or password else None
         )
+        logger.info("Password is : placeholder")
```

Comment:
We could log metadata about the password, e.g., is it an empty string or `None` or is it not?

Reply:
I'm aiming to review this for real in the next few days, but in light of issues like #671, I think something like this would be really helpful. I was wondering about printing the length, or showing asterisks, e.g.

```
>>> password = "my-awesome-password"
>>> print("Password: " + "*" * len(password))
Password: *******************
```

@sigmavirus24 or @di Any strong preferences on the UX here?

Reply:
I am strongly against revealing the length of a password. That's more metadata than is necessary.

Reply:
What about something like this?

```
>>> print(f"Password: <{'hidden' if password else 'empty'}>")
Password: <hidden>
```